pkgs/by-name/ni/nickel/package.nix

```diff
@@ -25,9 +29,18 @@ rustPlatform.buildRustPackage rec {
     "-p nickel-lang-lsp"
   ];
 
+  buildInputs = if enableNixImport then [
+    nix
+    boost
+  ] else [ ];
+
   nativeBuildInputs = [
     python3
-  ];
+  ] ++ (if enableNixImport then [
+    pkg-config
+  ] else [ ]);
+
+  buildFeatures = if enableNixImport then [ "nix-experimental" ] else [ ];
```

Comment:
How about `lib.optionals enableNixImport [ "nix-experimental" ];` (not sure about the actual function name, but I guess `lib.optionals` it is...

Same above of course.

Reply:
thanks, pushed to reflect that